### PR TITLE
pgr_depthFirstSearch and code review GSoC-2020 Week 9

### PR DESCRIPTION
**Pull request by krashish8**
Changes proposed in this pull request:
- Added a new DFS Visitor file
- Modified the implementation of `pgr_depthFirstSearch` for both directed and undirected graphs.
- Added extra edge-cases pgTAP tests to test the modified implementation, and modified the old pgTAP tests.
- Added pgTAP tests to test the equivalence of `pgr_depthFirstSearch` to `pgr_breadthFirstSearch` for the same `max_depth`, for some cases.
- Modified the docqueries for the modified implementation.
- Added the statement to catch cancelation from Postgres user
- Fixed minor styling issues and removed comments in the self-documented code.
- Renamed `source` / `start` to `root` everywhere.
- Removed unnecessary comments & modified the docstrings.
- Removed additional examples from the docqueries, added some examples for depthFirstSearch function.
- Fixed the named parameters ordering & documented the unnamed parameters in a proper way.
- Modified the pgTAP tests & the signatures file after ordering the named parameters properly.

@pgRouting/admins

### CODE REVIEW

File: doc/depthFirstSearch/pgr_depthFirstSearch.rst [hunk 1]
```diff
@@ -59,27 +58,34 @@ Signatures
...
 
 .. rubric:: Using defaults
 
 .. code-block:: none
 
-    pgr_depthFirstSearch(TEXT edges_sql, BIGINT start_vid)
+    pgr_depthFirstSearch(TEXT, BIGINT)
 
-    RETURNS SET OF (seq, depth, start_vid, node, edge, cost, agg_cost)
+    RETURNS SET OF (seq, depth, Root vid, node, edge, cost, agg_cost)
```
Comment (cvvergara) [inline doc/depthFirstSearch/pgr_depthFirstSearch.rst:73]: This is a column name: root_vid, only the IN compulsory parameters do not have names

File: doc/depthFirstSearch/pgr_depthFirstSearch.rst [hunk 2]
```diff
@@ -170,27 +176,27 @@ Result Columns
...
 Column           Type        Description
 ===============  =========== ====================================================
 **seq**          ``BIGINT``  Sequential value starting from :math:`1`.
 **depth**        ``BIGINT``  Depth of the ``node``.
 
-                             - :math:`0`  when ``node`` = ``start_vid``.
+                             - :math:`0`  when ``node`` = ``Root vid``.
 
-**start_vid**    ``BIGINT``  Identifier of the start vertex.
+**Root vid**     ``BIGINT``  Identifier of the root vertex.
```
Comment (cvvergara) [inline doc/depthFirstSearch/pgr_depthFirstSearch.rst:189]: There is something worng with the column name

**Comment by cvvergara** [comment]
The column name was ok with start_vid, as is the start vertex of the edge that is been traversted
trees have only one root

**Comment by cvvergara** [comment]
The OUT parameters all need to have names

**Comment by cvvergara** [comment]
Sequential coloring doc

that is a characterisitc of our results, its not a characteristic of the problem
```
The returned rows are ordered sequentially, starting from the node having the smallest value to the node having the greatest value.
```

**Comment by cvvergara** [comment]
Put this:
> The returned rows are ordered sequentially, starting from the node having the smallest value to the node having the greatest value.
as a characteristic of the results

**Comment by cvvergara** [comment]
> The implementation is applicable only for undirected graphs, because the direction of the edge has no significance to this algorithm.

Be punctual:
The implementation is applicable only for **undirected** graphs.
